spec/rails_helper.rb

```diff
@@ -7,6 +7,23 @@
 require 'rspec/rails'
 require_relative 'support/controller_macros'
 
+OmniAuth.config.test_mode = true
+omniauth_hash = {'provider' => 'github',
+                 'uid' => '12345',
+                 'info' => {
+                   'name' => 'test',
+                   'email' => 'test@github.com',
+                   'nickname' => 'test'
+                 },
+                 'extra' => {
+                   'raw_info' => {
+                     'location' => 'San Francisco',
+                     'gravatar_id' => '123456789'
+                   }
+                 }}
+
+OmniAuth.config.add_mock(:github, omniauth_hash)
```

Comment:
@sinpai what do you think about moving all Omniauth stuff to separate support module, located in `spec/support/omniauth_helper.rb`
```ruby
module OmniauthHelper
  def self.included(_)
    OmniAuth.config.test_mode = true
  end

  def mock_github_provider
    OmniAuth.config.add_mock(:github, omniauth_hash)
  end
  
  private

  def omniauth_hash
    {
      'provider' => 'github',
      'uid' => '12345',
      'info' => {
        'name' => 'test',
        'email' => 'test@github.com',
        'nickname' => 'test'
      },
      'extra' => {
        'raw_info' => {
          'location' => 'San Francisco',
          'gravatar_id' => '123456789'
        }
      }
    }
  end
end
```

In this case:
* we won't overload `rails_helper` file with logic
* will encapsulate separate support helpers

After those changes, you'll need to call `mock_github_provider` in `before` block in related tests

Reply:
@sinpai from time to time you'll add new support files, so there is a tool that can help to require all the stuff `require_all`

Reply:
don't forget to `config.include OmniauthHelper `